modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -958,7 +958,7 @@ public void setLocalPort(int locPort) {
      */
     @IgniteSpiConfiguration(optional = true)
     public void setLocalPortRange(int locPortRange) {
-        this.locPortRange = locPortRange;
+        this.locPortRange = locPortRange == 0 ? 1 : locPortRange;
```

Comment:
If the uses sets '0' as a range then getLocalPortRange() must return '0' as well. Please remove the modification at this part of the code and refer to my suggestions below.